src/main/java/com/telefonica/iot/cygnus/sinks/OrionPostgreSQLSink.java

```diff
@@ -0,0 +1,494 @@
+     * @return The PostgreSQL port
+     */
+    protected String getPostgreSQLPort() {
+        return postgresqlPort;
+    } // getPostgreSQLPort
+
+    /**
+     * Gets the PostgreSQL username. It is protected due to it is only required for testing purposes.
+     * @return The PostgreSQL username
+     */
+    protected String getPostgreSQLUsername() {
+        return postgresqlUsername;
+    } // getPostgreSQLUsername
+
+    /**
+     * Gets the PostgreSQL password. It is protected due to it is only required for testing purposes.
+     * @return The PostgreSQL password
+     */
+    protected String getPostgreSQLPassword() {
+        return postgresqlPassword;
+    } // getPostgreSQLPassword
+
+    /**
+     * Returns if the attribute persistence is row-based. It is protected due to it is only required for testing
+     * purposes.
+     * @return True if the attribute persistence is row-based, false otherwise
+     */
+    protected boolean getRowAttrPersistence() {
+        return rowAttrPersistence;
+    } // getRowAttrPersistence
+
+    /**
+     * Returns the table type. It is protected due to it is only required for testing purposes.
+     * @return The table type
+     */
+    protected TableType getTableType() {
+        return tableType;
+    } // getTableType
+
+    /**
+     * Returns the persistence backend. It is protected due to it is only required for testing purposes.
+     * @return The persistence backend
+     */
+    protected PostgreSQLBackendImpl getPersistenceBackend() {
+        return persistenceBackend;
+    } // getPersistenceBackend
+
+    /**
+     * Sets the persistence backend. It is protected due to it is only required for testing purposes.
+     * @param persistenceBackend
+     */
+    protected void setPersistenceBackend(PostgreSQLBackendImpl persistenceBackend) {
+        this.persistenceBackend = persistenceBackend;
+    } // setPersistenceBackend
+
+    @Override
+    public void configure(Context context) {
+        postgresqlHost = context.getString("postgresql_host", "localhost");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (postgresql_host=" + postgresqlHost + ")");
+        postgresqlPort = context.getString("postgresql_port", "3306");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (postgresql_port=" + postgresqlPort + ")");
+        postgresqlUsername = context.getString("postgresql_username", "opendata");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (postgresql_username=" + postgresqlUsername + ")");
+        // FIXME: postgresqlPassword should be read as a SHA1 and decoded here
+        postgresqlPassword = context.getString("postgresql_password", "unknown");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (postgresql_password=" + postgresqlPassword + ")");
+        String tableTypeStr = context.getString("table_type", "table-per-destination");
```

Comment:
In the meantime, we have added a general `data_model` parameter that is configured at `OrionSink` level. This new parameter may have values `dm-by-service`, `dm-by-service-path`, `dm-by-entity` and `dm-by-attribute`. `dm-by-service-path` matches `table-by-service-path`,  and `dm-by-entity` matches `table-by-destination`. Thus, this must be replace by a line in charge of reading the `data_model` parameter. The values `dm-by-service` and `dm-by-attribute` must be ignored.